src/ipc/ipc4/handler.c

```diff
@@ -1133,4 +1133,7 @@ void ipc_cmd(struct ipc_cmd_hdr *_hdr)
 
 		ipc_msg_send(&msg_reply, data, true);
 	}
+
+	dcache_invalidate_region((__sparse_force void __sparse_cache *)MAILBOX_HOSTBOX_BASE,
+				 MAILBOX_HOSTBOX_SIZE);
```

Comment:
@lyakh  This seems to have notable impact in large test runs. 

The one thing I don't fully understand is where are we writing to the mailbox. I only see reads from this memory, so not clear how can we have dirty data in the cache that would be written on top of whatever host writes to area.

ipc_msg_send() only writes to the registers.

Reply:
@kv2019i right, if we aren't writing there (and we shouldn't) then the DSP shouldn't mark the cache as dirty and therefore shouldn't write it back speculatively. So, indeed, probably it would be enough to make sure not to write there. Unfortunately, we do have a couple of locations where we write there. Probes do that, but they aren't used in our CI tests, so that wouldn't be the reason for failures. But I think I found more locations where that happens: in lines like
```
	config->frame_fmt = in_frame_fmt;
```
in copier.c. I'm working on further patches to fix that. If that's the only location, then fixing it should make this commit superfluous. We also need to check that we don't write caches back ourselves, and so far I don't see such cases.